FireboltNETSDK/Client/FireboltDataReader.cs

```diff
@@ -0,0 +1,473 @@
+            }
+        }
+
+        public override short GetInt16(int ordinal)
+        {
+            object value = ThrowIfInfinity(GetValue(ordinal), typeof(short));
+            switch (value)
+            {
+                case float f: return (short)f;
+                case double d: return (short)d;
+                case decimal d: return (short)d;
+                case byte b: return b;
+                case short s: return s;
+                case int i: return (short)i;
+                case long l: return (short)l;
+                case string s: return short.Parse(s);
+                default: throw new InvalidCastException($"Cannot cast ({value.GetType()}){value} to short");
+            }
+        }
+
+        public override int GetInt32(int ordinal)
+        {
+            object value = ThrowIfInfinity(GetValue(ordinal), typeof(int));
+            switch (value)
+            {
+                case float f: return (int)f;
+                case double d: return (int)d;
+                case decimal d: return (int)d;
+                case byte b: return b;
+                case short s: return s;
+                case int i: return i;
+                case long l: return (int)l;
+                case string s: return int.Parse(s);
+                default: throw new InvalidCastException($"Cannot cast ({value.GetType()}){value} to int");
+            }
+        }
+        public override long GetInt64(int ordinal)
+        {
+            object value = ThrowIfInfinity(GetValue(ordinal), typeof(long));
+            switch (value)
+            {
+                case float f: return (long)f;
+                case double d: return (long)d;
+                case decimal d: return (long)d;
+                case byte b: return b;
+                case short s: return s;
+                case int i: return i;
+                case long l: return l;
+                case string s: return long.Parse(s);
+                default: throw new InvalidCastException($"Cannot cast ({value.GetType()}){value} to long");
+            }
+        }
+
+        private static object ThrowIfInfinity(object value, Type type)
+        {
+            if (TypesConverter.isInfinity(value))
+            {
+                throw new InvalidCastException($"Cannot convert {value} to {type.Name}");
+            }
+            return value;
+        }
+
+        public override string GetName(int ordinal)
+        {
+            return _queryResult.Meta[ordinal].Name;
+        }
+
+        public override int GetOrdinal(string name)
+        {
+            return _queryResult?.Meta.FindIndex(m => m.Name == name) ?? throw new IndexOutOfRangeException($"Cannot find index for column {name}");
+        }
+
+        public override DataTable? GetSchemaTable()
+        {
+            return _fullTableName == null ? null : new DataTable(_fullTableName); // TODO split full table name
```

Comment:
Can you elaborate on this TODO? Nice to have or something that was missed?

Reply:
done